mixing/signatures.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2023-2024 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package mixing
+
+import (
+	"bytes"
+	"fmt"
+	"hash"
+
+	"github.com/decred/dcrd/crypto/blake256"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4/schnorr"
+)
+
+const tag = "decred-mix-signature"
+
+// Signed is an interface describing a signed mixing message.
+type Signed interface {
+	Pub() []byte
+	Sig() []byte
+	Sid() []byte
+	GetRun() uint32
+	Command() string
+	WriteSignedData(hash.Hash)
+}
+
+// SignMessage creates a signature for the message m and writes the signature
+// into the message.
+func SignMessage(m Signed, priv *secp256k1.PrivateKey) error {
+	sig, err := sign(priv, m)
+	if err != nil {
+		return err
+	}
+	copy(m.Sig(), sig)
```

Comment:
This works, but it's pretty wonky.  It's pretty rare for a func that returns something like this in an interface to actually write to it.  It's also assuming it's the right size and is rather brittle as a result.

It's fine for now, but it could probably use a comment that it needs to be revisited before a non v0 release version of the `mixing` module is created.